src/gui/dragAndDrop.js

```diff
@@ -2,48 +2,105 @@
  * Handles processing of drag and drop events for the mathematical Nodes
  */
 
+// Global imports
 /* global document */
 /* global NodeForest */
 /* global Calculator */
 /* global Operators */
 /* global NodeFactory */
+/* global NumberNode */
 
+/**
+ * Sets data attribute to hold the id of the picked up HTML Node
+ * @param event
+ */
 function pickUpNode(event) {
     event.dataTransfer.setData("text", event.target.id);
-    event.dataTransfer.setData("parent", event.target.parentNode.id);
 }
 
+/**
+ * Determines how to handle the dropping of each Node that is dragged into the editor space.
+ * @param event
+ */
 function dropNode(event) {
```

Comment:
This function is really large and should probably be divided.... I will not mark this as change needed unless something else is wrong though.